### DISCUSSION TITLE
Transferring data from one component to the next in Jatos

### DISCUSSION THREAD

**Discussion by jeremygrd** [jspsych/jsPsych#2682] [Q&A]
Hi,
since I'm using the latest version of jspsych where from what I understand there is no need for the onload syntax with jatos, 
I didn't find any example to save and retrieve data from one component to another. Using the jatos.setStudySessionData function is no longer as simple. If anyone has a clue, that would be fantastic! Thanks,
Best, 
Jeremy

**Comment by becky-gilbert** [comment]
Hi Jeremy! I'm not sure what you mean about not needing the `onload` syntax for JATOS anymore. I think you still need to use `jatos.onLoad` - this ensures that all of the JATOS functions have loaded before you try to use them. In the JATOS docs for jsPsych v7, they say to just put the `jsPsych.run` function inside `jatos.onLoad`, like this:
```js
// set up jsPsych experiment

jatos.onLoad(() => {
  jsPsych.run(timeline);
});
```
But I find it works just as well to put all of your jsPsych experiment set-up code inside the `jatos.onLoad` function:
```js
jatos.onLoad(() => {
  // set up jsPsych experiment
  jsPsych.run(timeline);
});
```

For transferring data between JATOS components, I would still use the study session data though I don't normally use the `jatos.setStudySessionData` function. Instead I write to and read from the [`studySessionData` object](https://www.jatos.org/jatos.js-Reference.html#jatosstudysessiondata) directly, like this:
**Component 1**
```js
// add a variable called 'id' to the session data
jatos.studySessionData.id = id;
```
**Component 2**
```js
// get the value of the 'id' variable that was added in Component 1
var id = jatos.studySessionData.id;
```
Does that work for you? Let us know if not and we can try to help figure it out!

**Reply by jeremygrd** [reply]
Thanks a lot for the quick answer! Indeed, I was playing each component separately and not testing the whole experiment. So it's not surprising that I couldn't get the variable I had just created. Sorry for the inconvenience.
Cheers

**Reply by becky-gilbert** [reply]
Ah yes, that's an easy mistake to make! Glad you figured it out 🙂
